src/features/MigrateLS.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState, useMemo, useEffect } from 'react';
+
+import { IFrame } from '@components';
+import { path, prop } from '@vendor';
+import { isVoid } from '@utils';
+import { getCurrentDBConfig } from '@database';
+import { TURL, LocalStorage } from '@types';
+
+const DBName = prop('main', getCurrentDBConfig());
+const getHostName: (obj: Window) => string | undefined = path(['location', 'hostname']);
+const getOrigin: (obj: any) => string | undefined = path(['contentWindow', 'location', 'origin']);
+const getLS: (obj: any) => LocalStorage | undefined = path([
+  'contentWindow',
+  'localStorage',
+  DBName
+]);
+
+const getIFrameSrc = (hostname: string) => {
+  switch (hostname) {
+    case 'mycrypto.com':
+      return 'beta.mycrypto.com' as TURL;
+    case 'mycryptobuilds.com':
+      return 'https://landing.mycryptobuilds.com' as TURL;
+    default:
+      return 'https://localhost:8000' as TURL;
+  }
+};
+
+const MigrateLS = ({
+  isEmpty = true,
+  importStorage
+}: {
+  isEmpty: boolean;
+  importStorage(ls: LocalStorage): boolean;
+}) => {
+  const [iframeRef, setIframeRef] = useState<HTMLIFrameElement>();
+  const [prevStorage, setPrevStorage] = useState<LocalStorage>();
+  const [error, setError] = useState(false);
+  const src = useMemo(() => getIFrameSrc(getHostName(window)!), []);
+
+  useEffect(() => {
+    console.debug('PrevStorage', prevStorage);
```

Comment:
Do we want to keep this ?

Reply:
Nope. It has been updated with better error handling.